src/core/InputText/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { TextFieldProps as RawTextFieldProps } from "@material-ui/core";
+import React, { ForwardedRef, forwardRef, useState } from "react";
+import { ExtraProps, StyledInputBase, StyledLabel } from "./style";
+
+interface AccessibleInputTextProps {
+  label: string;
+  placeholder?: string;
+  id: string;
+}
+
+export type InputTextProps = RawTextFieldProps &
+  ExtraProps &
+  AccessibleInputTextProps;
+
+const TextInput = (
+  props: InputTextProps,
+  ref: ForwardedRef<HTMLInputElement>
+): JSX.Element => {
+  const {
+    id,
+    intent = "default",
+    label,
+    placeholder,
+    sdsType,
+    hideLabel,
+    ...rest
+  } = props;
+
+  const [hasValue, setHasValue] = useState<boolean>(false);
+
+  const handleChange = (event: { target: { value: string } }) => {
+    if (event.target.value !== "") {
+      setHasValue(true);
+    } else {
+      setHasValue(false);
+    }
+  };
+
+  const inputProps = {
+    "aria-label": `${label}`,
+  };
+
+  if (!id || !label) {
+    // eslint-disable-next-line no-console
+    console.error(
+      `Error: czifui component InputText requires id and label props for accessibility.`
+    );
+    return <></>;
+  }
+
+  return (
+    <>
+      {!hideLabel && <StyledLabel htmlFor={id}>{label}</StyledLabel>}
+      <StyledInputBase
+        {...rest}
```

Comment:
should we move the `..rest` to the bottom to allow users to override the props 😄 ? For example, `type` could be `password`, `color`, `date`, etc..

https://www.w3schools.com/html/html_form_input_types.asp

Thank you!